packages/core-data/src/resolvers.js

```diff
@@ -193,107 +193,115 @@ export const getEditedEntityRecord = forwardResolver( 'getEntityRecord' );
  */
 export const getEntityRecords =
 	( kind, name, query = {} ) =>
-	async ( { dispatch } ) => {
-		const configs = await dispatch( getOrLoadEntitiesConfig( kind, name ) );
-		const entityConfig = configs.find(
-			( config ) => config.name === name && config.kind === kind
-		);
-		if ( ! entityConfig || entityConfig?.__experimentalNoFetch ) {
-			return;
-		}
-
-		const lock = await dispatch.__unstableAcquireStoreLock(
-			STORE_NAME,
-			[ 'entities', 'records', kind, name ],
-			{ exclusive: false }
-		);
-
-		try {
-			if ( query._fields ) {
-				// If requesting specific fields, items and query association to said
-				// records are stored by ID reference. Thus, fields must always include
-				// the ID.
-				query = {
-					...query,
-					_fields: [
-						...new Set( [
-							...( getNormalizedCommaSeparable( query._fields ) ||
-								[] ),
-							entityConfig.key || DEFAULT_ENTITY_KEY,
-						] ),
-					].join(),
-				};
+	( { dispatch, registry } ) => {
+		registry.batch( async () => {
```

Comment:
Question: does the resolver callback need to remain async? Does that mean batch needs to become async?
Question: could we build in batching resolver callbacks?

Reply:
Cc @youknowriad and @jsnajdr cause I don't really know what I'm doing here.

Reply:
This is not supported. This comment is relevant https://github.com/WordPress/gutenberg/pull/60495#issuecomment-2039309064

Reply:
I don't think batching async stuff make sense. You want to UI and meta selectors (isResolving) to actually update to show spinners... when requests are ongoing

Reply:
Yep, I understand now. I changed the PR to only batch the non async actions now.